### PR TITLE
option to hide/show domain in list of domains of a user 

### PR DESCRIPTION
**Pull request by sravfeyn**
This is not finished yet. A UI to hide/show needs to be added. 
For https://docs.google.com/a/dimagi.com/document/d/1H7rJFlr0-zqIEYA6JGrcFlA0GGtzzg7fu9vt-QzVfkM/edit#heading=h.vw1gk0mlqvik
The scope of this feature is just to Show/Hide - archiving will go through regular manual process.

### CODE REVIEW

**Comment by sravfeyn** [comment]
@czue hey, this is related to Project Archival feature. This is not finished, wanted to get your opinion on if this implementation looks okay, before going ahead with UI to set these settings.

File: corehq/apps/domain/models.py [hunk 1]
```diff
@@ -369,7 +369,7 @@ def active_for_user(user, is_active=True):
         else:
             couch_user = CouchUser.from_django_user(user)
         if couch_user:
-            domain_names = couch_user.get_domains()
+            domain_names = couch_user.get_domains(for_ui=True)
```
Comment (sravfeyn) [inline corehq/apps/domain/models.py]: Lot of places call `active_for_user`, will change this latter to pass `for_ui` up the stack once we decide what parts of UI should Hide domain-list and what parts should show all domains (even hidden from UI)

**Comment by czue** [comment]
seems ok to me. i think this feature becomes a lot simpler if we don't worry about archiving or deleting the data at all so makes sense for a first pass to just hide it.

File: corehq/apps/users/models.py [hunk 2]
```diff
@@ -378,6 +378,7 @@ class DomainMembership(Membership):
     role_id = StringProperty()
     location_id = StringProperty()
     program_id = StringProperty()
+    show_in_ui = BooleanProperty(default=True)
```
Comment (czue) [inline corehq/apps/users/models.py]: am sure you tested, but confirming this sets it to `True` when wrapping an object that doesn't have it coming out of the db?

**Comment by sravfeyn** [comment]
Thanks. Will go ahead with the UI
Yeah, pretty simple. Spent a lot more time scoping/discussing the feature than doing this easy_win.

File: corehq/apps/users/models.py (hunk 2 above)
Comment (sravfeyn) [inline corehq/apps/users/models.py]: Yeah, that's true.

**Pull request closed without merging**